text/0000-scoped-slots-light-dom.md

```diff
@@ -0,0 +1,227 @@
+by `<x-list>`, the child component. It's essentially like `<x-parent>` is slotting a function that actually
+returns the slotted content.
+
+This allows `<x-parent>` to control how each item renders while leaving the looping/repeating logic to `<x-list>`.
+`<x-list>` is now free to create as many slots as it wants while using the same "slotting function" passed from the parent.
+This allows `<x-list>` to be a virtual list or an infinite list.
+
+It's important to note that for this to happen, slots need to be lazily rendered.
+
+* Native shadow DOM slots: the slotted content is eagerly rendered regardless of there being a slot to “accept” the content. The component slotting the content is in charge of appending and removing the content from the DOM.
+* Scoped slots: the slotted content is lazily rendered; the slotted content will only be rendered if there is a slot to “accept” the content. The component with the <slot> elements is the one in charge of rendering the slotted content passed by the parent component.
+
+It is for this reason that Scoped Slots are allowed only in Light DOM components.
+
+## Detailed design
+
+Two new directives, `lwc:slot-bind` and `lwc:slot-data` are introduced that are used to bind data to the slotted content
+and to specify, in the parent, a placeholder to accept the bound data respectively.
+
+* ### `lwc:slot-bind`
+It has to be on a `<slot>` tag and should be an expression that is bound to a value within the child's scope. It can only
+be present on a Light DOM template. The compiler will throw an error if `lwc:slot-bind` is used in a shadow DOM template.
+* ### `lwc:slot-data`
+It has to be present on a custom element and should be a string literal that represents the variable name that will be
+used in the slotted content to reference the passed data. The custom element has to be a Light DOM LWC element. This check,
+though, will be performed at runtime because the LWC compiler lacks the necessary information to throw at compile-time.
+
+### Named Slots
+Accepting scoped slots for named slots presents a few logistical challenges illustrated below.
+```html
+<template>
+    <x-child lwc:slot-data="defaultSlotData">
+        <h1>Content that goes into default Slot {defaultSlotData.title}</h1>
+        <h2 slot="namedSlot">Named slot content {namedSlotContent.title}</h2> // where should namedSlotContent be declared?
+        <h3 slot="otherNamedSlot">{defaultSlotData.title}</h3> // `defaultSlotData` is actually out of scope, but *looks* in scope for a template author
+    </x-child>
+</template>
+```
+
+To address the above concerns, content slotted into named slots needs to be grouped together using a `<template>` tag
+and `lwc:slot-data` should be used on the `<template>`. 
+```html
+// x/parent.html
+<template>
+    <x-child> <!-- no `lwc:slot-data` on the component tag -->
+        <template slot="slotname1" lwc:slot-data="slot1data">

```

Comment:
This syntax makes sense for scoped slots. Would it makes sense to also allow this with standard slots?

Reply:
It's unclear whether the `Allowing slot attribute on <template> tag` section answers this question or not.

Reply:
Create a [section](https://github.com/salesforce/lwc-rfcs/blob/abdulsattar/scoped-slots/text/0000-scoped-slots-light-dom.md#implications-on-standard-slots) to explicitly call out that there will be no changes to standard slot authoring.